Aggressively check dotnet installed correctly.

JavidPack opened the pull request:
In discord support, a common issue is incomplete dotnet installs. Sometimes a dll is missing and the game crashes, sometimes dotnet.exe is missing. The cause of this is still unknown. Antivirus quarantines sounds like a reasonable reason, but users with this issue claim antivirus is off. Many users have success deleting the dotnet folder, or manually extracting the dotnet zip instead of letting the launch script do it.

This branch attempts to detect incomplete installs and force a reinstall if detected.


I'll get some #support people to test this script.

Solxanich commented on patches/tModLoader/Terraria/release_extras/LaunchUtils/InstallNetFramework.sh:
```diff
@@ -30,6 +30,16 @@ if [ -d "$dotnet_dir" ]; then
+	
+	# For some reason, sometimes the dotnet folder is missing some files, if so, delete it so it gets installed again
+	numberFilesInDotnetDir=`find "$install_dir" -type f | wc -l`
+	expectedFilesInDotnetDir=227
```
This number would be true strictly for windows x64 binaries. 

Example: Mac x64 binary is 185 files.

Do we want to restrict this to $_uname == _NT for windows only? 
Or do we want to make the fun switch case?

Rest of the code all looks ok, and seems to work fine.

Chicken-Bones committed:
1.4.4.2: Fix Utils.OpenFolder on windows

Chicken-Bones committed:
1.4.4.2: Preserve modded data when refreshing items (and reforging)

Chicken-Bones committed:
Update patches to 1.4.4.7

Chicken-Bones committed:
1.4.4.7: Remove TileLoader.MineDamage

- It had a bug which increased mining speed by 20%. Use power / modTile.MineResist directly instead.

Chicken-Bones committed:
1.4.4.7: Add tModPorter for TileID.Sets.TouchDamage changes

Chicken-Bones committed:
1.4.4.7: Add new player draw layers

Chicken-Bones committed:
1.4.4.7: Remove attack speed caps, and fix summon weapon speed patches

Chicken-Bones committed:
1.4.4.7: Cleanup bag opening hooks

JavidPack committed:
Add working directory to setup

JavidPack committed:
1.4.4.7: ExtractinatorUse changes, reimplementation, porter, and examples (#3085)

* ExtractinatorUse changes, reimplementation, porter, and examples

* Update PortingNotes_1.4.4.7.md

Chicken-Bones committed:
1.4.4.7: Use CommonMaxStack for all example items

Chicken-Bones committed:
1.4.4.7: Merge tML Item OnCreate hooks into new vanilla OnCreated

Chicken-Bones committed:
1.4.4.7: Better positioning of CanRightClick hook for openable items

Chicken-Bones committed:
tModPorter: Fix PlaySound rewriter with static Count field

Chicken-Bones committed:
tModPorter: Add not-yet-implemented markers so current tests can pass

Chicken-Bones committed:
Update TagIOTests so they all pass again

Chicken-Bones committed:
Update Roslyn to 4.3.1

- Decompiler dll was edited with dnSpy to fix breaking Humanizer version change

Chicken-Bones committed:
Switch to file scoped namespaces.

Chicken-Bones committed:
Fix collection initializer formatting

Mirsario committed:
Decompiler Update (#3083)

* Updated decompiler to '8.0.0.7106-preview2'.

* 'Terraria' patchstep.

* 'TerrariaNetCore' patchstep.

* 'tModLoader' patchstep.

* Editorconfig: Sort system directives first.

Mirsario committed:
Fixed vanilla step via a lost patch.

Mirsario committed:
Fixed offset of a single patch.

Chicken-Bones committed:
Remove old file from vanilla patch step

Mirsario committed:
File-scoped namespaces everywhere...

Mirsario committed:
Code Style Update (K&R, etc.) (#3097)

* Format Setup csproj files.

(cherry picked from commit 853f9dda5b4a64e68631ec45fd85d469b1fb1001)

* Improved csproj/targets generation with tabs & more.

* Switch to real K&R style - newlines on method declarations.

* Fixed indentation in case blocks

(cherry picked from commit de9a6fbdd6407df8966ad104bb2676f15a89838e)

* Re-undid a bad csproj change.

* 'Terraria' patchstep.

* TerrariaNetCore patchstep.

* 'tModLoader' patchstep done.

* Convert all missed methods, except non-patch TML files.

* Convert all remaining TML methods to K&R.

JavidPack committed:
1.4.4.7: DefaultToX usage in ExampleMod, Documentation (Take 2) #3088 (#3098)

* Reimplement code changes from #3088

* More fixes

Mirsario committed:
Merge branch '1.4' into 1.4.4

Mirsario committed:
Fixed crashes when deleting worlds/characters (Half of #3082)

Chicken-Bones committed:
Delete release_extras/Content folder from 1.4 merge

Chicken-Bones committed:
Delete old 1.4.3 porting notes

JavidPack committed:
1.4.4.7: Recipe shimmer examples and fixes (#3086)

* Shimmer recipe work, examples, comments, cloning

* Address feedback, move examples to ShimmerShowcase.cs

* Typo

* File-scoped namespace for the new file.

* Restore internalizing recipe bools

Co-authored-by: Mirsario <me@mirsar.io>

Chicken-Bones committed:
1.4.4.7: Remove some outdated NPC profile related code.

Chicken-Bones committed:
Fix failing tModPorter test after 1.4 merge

Chicken-Bones committed:
Remove nullable bool? from Mod/GlobalNPC.CanHitNPC

Chicken-Bones committed:
1.4.4.7: Update WorldGen.Convert documentation

Chicken-Bones committed:
1.4.4.7: Update/Rename hook, ModNPC.ApplyDifficultyAndPlayerScaling

Chicken-Bones committed:
Remove CanBurnInLava hook, as suitable replacements exist.

Chicken-Bones committed:
Change HookGen namespace style from IL/On. to IL/On_ on the type name

Chicken-Bones committed:
1.4.4.7: Add tModPorter comment to ApplyDifficultyAndPlayerScaling

Chicken-Bones committed:
Forgot to commit latest version of TerrariaHooks

Chicken-Bones committed:
Refactor a bunch of hook callsites into CombinedHooks

direwolf420 committed:
Fix tile framing (causing infinite UnloadedXTile loop and invisible tiles) (#3107)

* tile frame fix

* Minor fix to formatting after merge conflict fix

* Missed a line removal in merged conflict resolution

Co-authored-by: Solxanich <59670736+Solxanich@users.noreply.github.com>

Mirsario committed:
Patch improvements (#3105)

* Insane full patch review.

* Undo accidental non-cs reverts.

* Fixed asset (& mod icon) issues.

* Undo useless change.

* CB review of Terraria/TerrariaNetCore & tML ReLogic patches

* Strip UTF8 BOM from cs files

* CB review of tML patches

Co-authored-by: Chicken-Bones <mehvids@gmail.com>

Chicken-Bones committed:
1.4.4.9: Vanilla UIElement member refactor

Chicken-Bones committed:
Update patches to 1.4.4.9

Chicken-Bones committed:
Merge remote-tracking branch 'origin/1.4' into 1.4.4

direwolf420 committed:
[1.4.4] Change return type of Mod.AddContent to bool, add docs (#2953)

* change return type of Mod.AddContent to bool, add docs

* invert to avoid branching

* existing doc unification/clarification

* inheritdoc usage to capture doc changes

* fix bad merge

Chicken-Bones committed:
1.4.4: Fix rendering of Mod Configuration ingame menu option

Ved-s committed:
Ru localization (#3108)

* Ru localization

* Apply suggestions from code review

Co-authored-by: Mirsario <me@mirsar.io>

Solxanich committed:
Fixes Mod Tree Sapling placement (includes axe of growth) (#3118)

* Fixes Mod Tree Sapling placement (includes axe of growth)

* Update Porting Notes

Rijam committed:
[1.4.4] Obsidian Armor Set Bonus Whip Speed Fix (#3121)

* Obsidian armor whip speed fix

* Alternate fix

JavidPack committed:
Typo in ModifyHardmodeTasks docs

Jadams505 committed:
Fix layer drawing (#3124)

direwolf420 committed:
[1.4.4] Boss health bar update (#3117)

* Boss Health Bar updates
* Implement showText functionality
* Update comments for proper formatting
* Make public DrawHealthText method

* move DrawHealthText into partial

* misc ExampleMod updates

* comment

Mirsario committed:
Change DisplayNameInternal properties to 'private protected'.

Mirsario committed:
Workaround for window maximizing issues (Fixes #3082)

Mirsario committed:
Move & rename ModItem.SacrificeTotal (#3078)

* Moved ModItem.SacrificeTotal to Item

* ...Actually do what the previous commit said.

* Defense for the getter.

* Rename to ResearchUnlockCount.

* Minimize using changes.

* Leftovers.

* Add tModPorter refactor for SacrificeTotal -> Item.ResearchUnlockCount

Co-authored-by: Chicken-Bones <mehvids@gmail.com>

direwolf420 committed:
fix PrefixID.Sets not being resized (#3131)

direwolf420 committed:
more ModBiome/SceneEffect docs, changes (#3135)

direwolf420 committed:
[1.4.4] Log worldgen seed & consistency with existing world loading messages (#3130)

* log worldgen seed & consistency with existing world loading messages

* consts

* fix last Size=, change IsExpert to GameMode

Jadams505 committed:
Update vanillaSets with 1.4.4 options (#3136)

Jadams505 committed:
Visual chest transfer fix (#3129)

Sneakilli committed:
Added catch conditions for ExampleQuestFish (#3089)

* Added catch conditions for ExampleQuestFish

This fish could previously not be caught

* Update ExampleMod/Common/Players/ExampleFishingPlayer.cs

Co-authored-by: direwolf420 <amakejk1@web.de>

* Removed extraneous checks when catching ExampleQuestFish

Co-authored-by: direwolf420 <amakejk1@web.de>

Cyrillya committed:
[1.4.4] Infinite shooting fix (#3139)

* Set `itemAnimationMax` to -1 when entering world

* check `context == PlayerSpawnContext.SpawningIntoWorld`

* add an `itemAnimation > 0` && check to `ItemAnimationJustStarted`

* Move -1 comment into right spot

Co-authored-by: Chicken-Bones <mehvids@gmail.com>

Jadams505 committed:
[1.4.4] Remix seed fix (#3146)

* Remix seed generation fix

* Clear tile data

Cyrillya committed:
Fix quick heal & mana consuming non-consumable items (#3063)

* Add `item.consumable` check

* update 1.4.4

Cyrillya committed:
Call `ItemLoader.UseItem` for food quick buff (#3151)

Chicken-Bones committed:
Delete old 1.4.4.7 porting notes

Chicken-Bones committed:
Fix crash when missing the old local Content/Images override directory

Setnour6 committed:
Initial Condition Adding Commit (#3165)

direwolf420 committed:
[1.4.4] Add ItemID.Sets.GeodeDrops (#3145)

* ItemID.Sets.GeodeDrops set

* move into separate method, fix exception text

* move validation into load step

PaulHause committed:
[1.4.4] WIP: Reworked Biome Block Counting for Vanilla Biomes (#3168)

* * Rebased to 1.4.4

* Added new tile sets to accomdate remix worlds

* Restructured tile count code to be less repetetive and account for remix.

* Added 2 functions to help simplify modders adding blocks to biomes regardless of remix world.

* * Added DungeonBiome set to allow for dungeon compatability.

* Renamed sets for consistency.

* Turned certain summaries into comments to avoid bloat from self-explanitory objects.

* Optimized RecountTiles by removing unnecessary checks and cleaning up code.

* * Moved AddXTile functions from TileLoader.cs to TileID.TML.cs

Chicken-Bones committed:
Fix #3169

Freedbot committed:
Made server manage script curl/wget agnostic (#3081)

* Made server manage script curl/wget agnostic

Script will silently attempt to use curl first.  Failure is silent, then wget is silently attempted.  Failing wget is not silent, so failing both will produce a wget error.
Further, check_update and copy_worlds functions are condensed for simplicity.
Since the functionality is basically unchanged, version increment is minimal.

* Update manage-tModLoaderServer.sh

Complying with Solxan's suggested format fix.

direwolf420 committed:
[1.4.4] Add TileID.Sets.AvoidedByMeteorLanding (#3162)

* Add TileID.Sets.AvoidedByMeteorGeneration

* ExamplePylon uses new set

* formatting

* doc fix

* rename to Landing

Chicken-Bones committed:
Fix tModPorter refactors not being performed on obsolete MemberReference operations

Chicken-Bones committed:
Fix tModPorter crash when target type of an invoke refactor cannot be determined

Chicken-Bones committed:
tModPorter: SendObjectPlacment->SendObjectPlacement

Chicken-Bones committed:
Fix AssemblyManager.ForceJITOnMethod() Throwing on Interface Methods (#3175)

* force method JIT logic to not check "virtual" methods in interface types

* `is false` instead of `== false`

* use `is not true` instead of `is false` to allow methods with null DeclaringType to be evaluated

* `is false` was actually better

Solxanich committed:
Trailing cleanup of start server.sh. Fixes #3137

Chicken-Bones committed:
Improve Player.clientClone performance (#3174)

* Draft of CopyNetStateTo and IsNetStateEquivalent

* Use CopyNetStateTo in ModAccessorySlotPlayer

* ModPlayer.(clientClone -> CopyClientState) rename & redocumentation.

* Add tModPorter

* Throw exception in Item.Clone when bad for performance

Co-authored-by: Mirsario <me@mirsar.io>

JavidPack committed:
Port #3128 to 1.4.4

JavidPack committed:
Fix UIModSourceItem overlapping buttons

Solxanich committed:
Port b0e0b1de59a2f2321025b144b32669614023ea54

Solxanich committed:
Merge branch '1.4.4' of https://github.com/tModLoader/tModLoader into 1.4.4

Solxanich committed:
Port b0e0b1de59a2f2321025b144b32669614023ea54

Chicken-Bones committed:
Remove unnecessary Player parameter from some ModPlayer hooks

Solxanich committed:
Port dd701488a8c471f1cba01065d0600eb875f3664a

Solxanich committed:
Merge branch '1.4.4' of https://github.com/tModLoader/tModLoader into 1.4.4

Solxanich committed:
Port: Fixes #3032 (#3171) by  Solxan

Solxanich committed:
Port: Fix Content Folder Path On Mac (#3173) By ilcheese2

Co-Authored-By: ilcheese2 <56899583+ilcheese2@users.noreply.github.com>

Solxanich committed:
Port: Vastly Speed Up Publishing Menu By Solxan

Solxanich committed:
Port: Ignore tml patch version for Mod Exception By Solxan

Mirsario committed:
Scaled down the TML logo in Doxygen.

(cherry picked from commit ad8b59eba69a83daeda44b66e0688df8bf5bee88)

Chicken-Bones committed:
Fix #3179 - Specific attack speed multipliers causing animation and use time desync

Cyrillya committed:
[1.4.4] `ModPlayer.AddMaterialsForCrafting` Method (#3183)

* ModPlayer.AddMaterialsForCrafting method + Example Mod usage

* Add the cloned version of the item to `ConsumedItems` list

* Change return type to `IEnumerable<Item>` and (tried) to remove auto-formatter whitespace changes

* Rework hook implementation to have less allocations.

* Parameter renaming + Locked chest check for the example

Co-authored-by: Chicken-Bones <mehvids@gmail.com>

Kogsey committed:
Shimmer Recipe.Condition support + internalisation/removal of vanilla corruption/crimson versions (#3193)

* Added basic functionality

* Added multi-recipe support to decrafting, now shimmer loops through all recipes that craft an item and their conditions to find one that is available

Also added two new recipe conditions for crimson and corruption worlds for ease of use, obsolete tagged the old methods, renamed my methods to Decraft over Shimmer because it felt better and updated example mod

* Removed a pesky exclamation mark, learned how arrays work, and added a missed else statement

Just fixes to make it work, tested with corruption, crimson, and desert conditions, last entry successfully takes priority, currently in working state

* Added EnableShimmer method, priority integer for shimmer transforms, and appropriate examplemod code

Added an EnableShimmer method to match DisableShimmer with summary on both explaining they are only used during load, SetShimmerPriority(int) allows manual priority changes, still following the last added first priority method for duplicate integers

* Changed recipe to num in ShimmerTransforms, removed obsolete functions, and fixed lines as per Dire's request

* Somehow did not actually change recipe to num on last commit

Do not code late at night

* Changed Decraft in methods back to Shimmer, fixed a json mistake, setup auto-replace for Recipe.crimson and Reciple.corruption then removed ItemID.Sets.IsCraftedCorruption/Crimson as they were then unused

* Removed priority system, renamed multiple methods under shimmer to decraft, added crimson and corruption flags for battle potions and the mechanical worm, made UpdateWhichItemsAreCrafted() use lists instead of arrays, mostly as per Chicken-Bones' request

Updated examplemod and comments appropriately

* Fixed two comments refering to DisableShimmer() instead of DisableDecraft()

* Removed comment in ItemID,  removed unnecessary check in ShimmerTransforms.GetDecraftingRecipeIndex(), and moved Recipe.notDecraftable to original position in line with vanilla

Kogsey committed:
Fix for multiplayer modded music boxes not working with wiring (#3191)

* Fix for multiplayer modded music box wiring support

Revert "Fix for multiplayer modded music box wiring support"

This reverts commit 9bbad6fa7971c20b4f1437049419aec86be9551d.

Revert "Revert "Fix for multiplayer modded music box wiring support""

This reverts commit 1a47a581f579fd7ea683574aad6f728aff14466b.

Added a note to the variable in Main

Just for documentation, prev reverse was because I cannot use my mouse

Moved tileIsModMusicBox from Main to TileID.Sets and made its comment descriptive as per Dire's request

Public variables are overrated - as per Dire's request

* Chicken Bones' simpler fix, allowing MusicLoader.AddMusicBox to pass a value of 0 through to musicSlot and into the dictionaries

This reverts commit 38572da9794851bce77dda88881dee2d6aff85e2.

Destructor-Ben committed:
UIMessageBox Fix (#3190)

* UIMessageBox Fix

Fixed an incredibly minor error with the UIMessageBox, where text was not cut off properly while scrolling.

* Fix width resizing, scroll resetting, efficient

Co-authored-by: JavidPack <javidpack@gmail.com>

JavidPack committed:
Measure ModMenu display name correctly (#3111). Ported from 1.4

Co-Authored-By: Jomar Milan <macaylamarvelous81@proton.me>

JavidPack committed:
Message uistate visual consistency.

JavidPack committed:
Fix #3027 , don't allow illegal mod names in Create Mod menu.

Solxanich committed:
Update build.yml

Solxanich committed:
Update build.yml

direwolf420 committed:
fix DustType + CreateDust interaction (#3200)

direwolf420 committed:
reset rarity for popup text (#3201)

direwolf420 committed:
reorder ClearEdges to clear modded tiles (#3203)

Chicken-Bones committed:
tModPorter - Fix crash when RecipeRewriter encounters a null literal op

Chicken-Bones committed:
Require exact .NET major version for mod development (tModPorter)

Jadams505 committed:
Fix sign cursor spam (#3208)

JavidPack committed:
1.4.4 localization overhaul (#3101)

* Initial Localization file updating implementation.

* ExtractLocalization button implementation

* hjson prefix implementation

* Fix comment indentation, fix first root comment

* Address feedback, implement flattened output if only 1 entry, skip some work if possible.

* Refactor LocalizationLoader with more cleanups

* Move GameCulture.FromPath extension into LocalizationLoader

* More naming refactors

* Initial work towards ModTranslation class removal.

* ContainerName experiment, compilable again

* Cleanup implementation of LocalizedText properties and creation

* Halfway through patron localization entries

* Restore ModBlockType.CreateMapEntryName

* Change Category to LocalizationCategory

* Pluralize categories

* tModPorter for localization changes. Applied to Example Mod

* Rename GetOrAdd to GetOrRegister. Add tModPorter for removed LocalizationLoader methods

* Update `LocalizedText.Value` instead of replacing instances when loading mod translations

* Hjson files from 1.4.3 export

* Fix ExampleChest, Prefixes typo

* Unload modded keys by reloading language. Move RefreshModLanguage to LoadLanguage

* tModPorter: SendObjectPlacment->SendObjectPlacement

* Remove commented-out ModTranslation.SetDefault calls, comments in hjson.

* Migrate more ModLoaderMod localization

* Localization file update

* GetLocalizedValue method for streamlined localized text retrieval.

* Add mod last modified check to UpdateLocalizationFilesForMod

* Scope-aware localization key substitutions, fix other issues

* Update pluralization code. Russian/Polish translations may need tweaking

- Also removed LocalizedText.Override

* Refactor Scope-aware localization key substitutions code

* Fix issue with extracting localization files from mod

* Rename GetOrRegisterLocalization to GetLocalization

* Add LocalizedText.WithFormatArgs

* Trigger mod rebuild in VS on .hjson file modification

* Properly localize ExampleHealingPotion

* Remove dead ModTranslation loading code

* Simplify and improve language loading (including resource packs)

* Add mod name to any hjson loading exceptions

* Restore game tip autoloading

* Autoload translation keys in the default culture

* translationMod feature, inherit localization entries from other mods.

* Add CommonItemTooltip entries and substitution offsets

* Only update hjson if mod in mods folder

* Use // for single line comments in hjson files

* Cleanup redundant logic in Russian and Polish plural rules

* Remove stale comment

* 1.4.4 localization overhaul file watcher (#3207)

* Initial hjson live reload via filesystemwatcher

* Update feature to support multiple mods, multiple files at same time, feedback

* Reorganize to LocalizationLoader, use lock.

* Cleanup

* Lock on collection itself and mark collections as readonly

Co-authored-by: Mirsario <me@mirsar.io>
Co-authored-by: Chicken-Bones <mehvids@gmail.com>

Solxanich committed:
Family Share Manual Launch Script (#3211)

* Family Share Manual Launch Script

* Add Unix launch script equivalent

Solxanich committed:
Port Better Preview Warnings #3178

JavidPack committed:
Update CI for 1.4.4.9

Solxanich committed:
Try auto-build 1.4.4

Solxanich committed:
Fix the build script to be more versatile

JavidPack committed:
Autopublish 1.4.4 to Steam: 1.4.4-preview

Solxanich committed:
Updated Github Actions for 1.4.4 Testing

Solxanich committed:
Disable Date block in PreviewChanges

Solxanich committed:
Dummy Push for fixing tagging

Solxanich committed:
Fix bug with non-removed 1.4.4 in actions

JavidPack committed:
Fix bad en-US filename issues, enable orphan file rename, fix a dup key issue

JavidPack committed:
Adjust BuildInfo logic for new branch names

absoluteAquarian committed:
OnStack API Documentation Improvements (#3225)

* cleaned up the OnStack API

* ExampleMod cleanup

* forgot the GlobalItem example

* reverted `as` to casting, XML summary changes

* typo fix

* changed summary for TransferWithLimit

* more variable renaming

* prefer `if` statements over `??=`

* forgot to change ModifyTooltips

* revert `null` usage to `string.Empty`

* added brackets for `if` statements

* more XML summaries

* typo fix and more XML summaries

Saethar committed:
Saethar patron set changes & fixes (#3224)

* Add files via upload

* Delete Saethar_Body_Arms.png

* Add files via upload

* Modernized Saethar.cs

* Eye glow rendering.

* Fixed wobbling in this & Guildpack's glow layers.

* Only require the hat for the eye glow..

* Fixed away artefacts.

---------

Co-authored-by: Mirsario <me@mirsar.io>
(cherry picked from commit 0a061eea93cf1c7c50cf6855f2fdebcd3de4fa0c)

JavidPack committed:
Fix #2835 , fix swapping platform under modded chests crashing or breaking chest.

JavidPack committed:
Fix TMLVERSION preprocessor directive missing

JavidPack committed:
Fix Create Mod output for localization changes, open output folder automatically.

JavidPack committed:
Move recent fix to more sensible location.

JavidPack committed:
Fix #1639 . Fix issue with too many wings. Change all remaining Item equip slot fields from sbyte to int.

vksir committed:
fix -nosteam problem (#3217)

Solxanich committed:
Readjust vanilla Achievement code patches

ppebb committed:
Fix manage script not moving mods if --check-dir is set (#3232)

lace-wing committed:
1.4.4 localization entries Chinese translation (#3230)

Mirsario committed:
Fixed achievement chat tags not being initialized.

Jadams505 committed:
[1.4.4] Modded chest lock support (#3123)

* Fix chest lock bug

* LockChest hook

* Update ExampleMod

* More detailed ExampleMod implementation comments

* Update TileLoader.cs

---------

Co-authored-by: JavidPack <javidpack@gmail.com>

JavidPack committed:
Fix #3231 (Fix ModifyScreenPosition), add CameraModifier example to MinionBoss

JavidPack committed:
Fix #3055

JavidPack committed:
Documentation for NPC buff methods

direwolf420 committed:
reimplement ExampleBlock.ChangeWaterfallStyle (#3244)

direwolf420 committed:
GrappleCanLatchOnTo hook (#3248)

* add GrappleCanLatchOnTo hook

* example comments

* simplify nullable call

direwolf420 committed:
PostSetupTileMerge hook, fixes related to tile merging (#3243)

* PostSetupTileMerge hook, fix modded BlockMergesWithMergeAllBlock assignment

* rename TileLoader method, move closer to intended context

* TileID.Sets.BlockMergesWithMergeAllBlockOverride

* address feedback

JavidPack committed:
Remove unnecessary code, fix some research counts, default to 1 (#3242)

JavidPack committed:
Fix #3253

JavidPack committed:
Prevent issue with localization files having changes all the time despite no changes.

JavidPack committed:
Fix #3247 (Fix recipe group regressions), more documentation

JavidPack committed:
AddMapEntry Docs

Mirsario committed:
Updated .NET versions to 6.0.14 (6.0.406)

Mirsario committed:
HookGen!

RazzSG committed:
Improve ru-RU tModLoader.json translation (#3264)

* Improved ru-RU tModLoader.json translation

* Update patches/tModLoader/Terraria/Localization/Content/ru-RU/tModLoader.json

Co-authored-by: Mirsario <me@mirsar.io>

* Update patches/tModLoader/Terraria/Localization/Content/ru-RU/tModLoader.json

Co-authored-by: Mirsario <me@mirsar.io>

---------

Co-authored-by: Mirsario <me@mirsar.io>

Cyrillya committed:
Fix writing extra buff slots to vanilla buffer causing reading beyond the end of the stream (#3263)

direwolf420 committed:
Fix items falling through tiles and certain projectiles updating in non-loaded chunks in MP (#3245)

* fix items falling through tiles in non-loaded chunks in MP

* golfball/chum projectile fix

direwolf420 committed:
Better netcode error messages for Globals (#3274)

direwolf420 committed:
Fix EM not loading on servers, ExampleRecipeMaterialPlayer fixes (#3272)

direwolf420 committed:
Fix ProjectileLoader.ReceiveExtraAI not being called (consistently) (#3273)

* fix ProjectileLoader.ReceiveExtraAI not being called consistently

* cleaner solution to flagging

direwolf420 committed:
ExampleMod: More shimmer transformations (#3262)

direwolf420 committed:
Fix pet drawing in player select screen for invalid items + ExampleMod showcase (#3257)

* fix pet drawing in player select screen for invalid items

* ExampleMod showcase of CharacterPreviewAnimations, update some comments

* typo

* ExampleMod: update purpose

* ExampleMod: comment tweaks

* ExampleMod: fix comment

* ExampleMod: fix comment again

direwolf420 committed:
Fix bugs from throwing fields port (#3238)

* Fix bugs from thowing field backports:
* in SP, landing falling stars would not "drop" the item if a thrown consumption prevention effect was enabled
* thrown consumption prevention effects would not work for non-ammo style throwing items

* use correct bool value

* more noDropItem docs

Chicken-Bones committed:
Replace usages of as with casts or pattern matching in ExampleMod

Setnour6 committed:
Add `NearShimmer` recipe condition (#3277)

* Add NeedsShimmer recipe condition

* Add `adjShimmer != oldAdjShimmer` flag condition

* Typo fixes, flag changes

* Undo `Recipe.cs` patch changes

direwolf420 committed:
InfoDisplay text coloring support (#3258)

* Add support for InfoDisplay text coloring

* ExampleMod, fix bad LINQ usage

* tModPorter

* ExampleMod tweaks

* inactiveColor is now static property

JavidPack committed:
Update NewProjectile docs

JavidPack committed:
Better error messages for malformed hjson

Chicken-Bones committed:
Fix localization format arg offset substitution regex

Chicken-Bones committed:
Apply pluralization to bound localization format args

lace-wing committed:
"Needs shimmer" and unloaded chests Chinese translations (#3280)

* Needs shimmer zh translation

* Unloaded chests zh translation

Mirsario committed:
Fixed #3281 (QS mount visual issue)

Chicken-Bones committed:
Update NuGet.Frameworks reference to work with latest VS

Chicken-Bones committed:
Set tModPorter to invariant culture to help with MSBuild error messages

direwolf420 committed:
Fix 1.4.4 specific bugs caused by vanilla ID count restrictions (#3246)

* Tile, Wall, NPC ID vanilla count to modded

* remove NPC achievement patch

* remaining IDs

* remove Count checks where reasonable

MutantWafflez committed:
Completed Several Tasks From `PortingNotes_1.4.4.9.md` (#3287)

* PrefixLoader.GrantPrefixBenefits

* Update todo list

* Replace SingleGrappleHook with Set

* QuickHeal/Mana already done

* Make FindOwner code use Player.GetItemGrabRange

* MaxBuffs rename already done

* Add hook for `RefreshInfoAccsFromItemType`

* Update ArmorPenetration values in SetDefaults

* TownNPCProfiles w/ Shimmer in ExampleMod

* Replace Door IDs in ModTile with Sets

* Singular comma removal

* Comment Related Changes

* tModPorter stuff (Not 100% with this)

* More tModPorter-ing

* Requested changes made

* Remove unnecessary shimmer checks

* 2nd requested changes made

* Ensure loadouts don't save modded Item IDs

Chicken-Bones committed:
Fix bad write implementation from #3287

MutantWafflez committed:
Modded Loadout IO (#3290)

* Modded Loadout IO

* Use array length rather than magic number

* Remove unnecessary TryGet check

Antirhinnum committed:
[1.4.4] A Little* Bit of Documentation (#3251)

* Initial Commit

* Added a spoonful of documentation.

* Fix Accidental `Chest` Patch Reversion

* Un-reverted a change from 48f0a29d26cc916ae8e89f061ba32017f9af9a15.

* Change most ID references

* Change most references to ID classes (such as `ItemID`, `NPCID`, etc.) to a more general "<x> type" comment. ID classes are still referenced if they make more sense (for example, the persistent name dictionaries in `ContentSamples`).
* Fixed several minor spelling mistakes.

Buff IDs and equip IDs are not yet changed.

* Add some more info

* Fix patches from merge

---------

Co-authored-by: JavidPack <javidpack@gmail.com>

Exiled1 committed:
Resolves #3133: added a currency system accessor (#3300)

This change simply adds a new hook to try to get a currency system given
ID

Exiled1 committed:
Fixes #3221 by adding a check to player and world names to prevent reserved keywords from being used.  (#3298)

* Resolves #3221

There was a bug with map files having the potential to create files
named as reserved keywords/files in Windows, which causes the file to
become incredibly difficult to remove.

This patch simply adds a check to the world and player names to check if
they have any of these reserved items and appends a '_' to them.

* Fixed an accidental misspelling of a variable

* Fixed formatting issues to follow the style guide

Chicken-Bones committed:
Minor formatting fixes

MutantWafflez committed:
New 1.4.4 Info Accessory Hooks (#3292)

* ModPlayer hooks

* Mod/GlobalItem Hooks

* ExampleMod & Comment Updates, call order fix

* Remove task from todo list

* Resprite of ExampleInfoAccessory (by Rijam#3788)

* Request Changes Made

- ExampleInfoAccessory now works in Void bag w/ comments
- Made vanilla RefreshInfoAccsFromItemType(int accType) private. shadowed by tML method

BasicallyIAmFox committed:
[1.4.4] Add ItemID.Sets.OreDropsFromSlime (#3295)

* added OreDropsFromSlime item id set

* Update ExampleOre.cs

* Not so minimal patches

* new

* Update NPC.cs.patch

* Update NPC.cs.patch

* Update NPC.cs.patch

* Update NPC.cs.patch

Chicken-Bones committed:
Fix failing tModPorter tests

Chicken-Bones committed:
Fix some armor pen being applied to the wrong projectile

Rijam committed:
[1.4.4] Updated Shimmered Example NPCs and Added Gore (#3172)

* Sprites, Set, ITownNPCProfile, comments

* Added Traveling Merchant and Bone Merchant

* Updated Profiles & Added Gore

* Removed unused using statements

* Removed extra unused textures

* Update the gore retrieval code, support party hats, UsesPartyHat example

* Simpler syntax

---------

Co-authored-by: JavidPack <javidpack@gmail.com>

Chicken-Bones committed:
Add a scoped disposable to silence the logging FirstChanceExceptionHandler

Chicken-Bones committed:
Allow pluralization to work on non-integers (by converting them to ints)

JavidPack committed:
Fix bounds in #3295

JavidPack committed:
Fix #3303 , move DrawBackHair to HairID

Chicken-Bones committed:
Player/NPC damage hooks rework. Hit/HurtModifiers and Hit/HurtInfo (#3212)

* Add DamageStrike struct for damage dealt to NPCs

* Update hooks for new DamageStrike parameter

* Restore ichor and betsysCurse debuffs

* Override damage dealt for instant kill arg

* Draft 2 (still untested). Projectile.ModifyDamageScaling needs work

* Documentation feedback

* Get all the stat modifier chains right for Projectile.Damage (probably)

* Reduce patches for Player.ProcessHitAgainstNPC damage multipliers

* Expand StrikeModifiers with dmg variation, bonus damage, and more

* Add the ability to scale damage variation

* Rename DamageStrike -> Strike and move to ModLoader namespace

* Rename ModifyStrike hook names to ModifyHit for consistency

* Add AddableFloat and MultipliableFloat to ModifyCollisionData and ModifyWeaponCrit hooks

* Upgrade Player.Hurt with hooks, HurtModifiers, HurtInfo

* Rename StrikeModifiers -> HitModifiers and Strike -> HitInfo

* Fix hit serialization

* Move HitModifiers/Info into NPC as inner classes

* Missed rename refactors, and some doc fixes

* Add DefenseStat for player defense

* Revert ModifyWeaponCrit changes. They belong in another PR

* Fix tModPorter BaseRewriter.UseType with qualified names

* Reduce to a single ModPlayer.OnHitNPC hook.

* Add some extra net safety to PlayerDeathReason.TryGetCausingEntity

* Add NotNullWhen to Projectile.TryGetOwner

* Restore Modify/OnHit Item/Projectile convenience hooks in ModPlayer

* Remove Projectile.ModifyHit (originally ModifyDamageScaling)

* Make HurtModifiers and HitModifiers easier to construct.

* Fully support HitDirection and HitDirectionOverride on both NPC and Player hits

* Add DamageType to NPC.HitModifiers

* tModPorter!

* Remove incredibly hard to maintain vanilla compat netcode

* Straightforward porting

* documentation work/fixes

* No 100% certain correct damage StatModifier used

* Small fix

* Fix Crit damage being 100% instead of 200%

* WeaponWithGrowingDamageToggle, default to false

* Example of projectile doing extra damage to specific NPC

* example of an enemy which is resistant to magic damage

* Dodge, paladin shield, HitModifiersShowcase

* Code style fixes, prevent Main.NewText erroring on server

* Update whip tag damage, add buff textures

* Account for SummonTagDamageMultiplier, documentation

* Consistent Percent and Multiplier usage

* Fix Crit damage being 100% instead of 200%

* Fix StrikeNPC packets bouncing back to sender

* Fix projectiles lacking damage variation

* Little refactors, typos for paladin shield code

* Fix some bad xml

* Projectile.IsMinionOrSentryRelated helper, CommonItemTooltip.SummonTagDamage key usage

* Added docs about which modify/on hit methods are called on clients/server

* Minor documentation adjustments

* Improve implementation/documentation of paladin's shield example

* Fix some spacing

* exampleDodgeCooldown match documentation

* Add an extra note about the net benefits of using buffs

* Address feedback

* Fix double-hit messages in PvP

* Sync changes for showcase items, fix a comment

* Comment, docs

* Fix players receiving no knockback

* ModifyHitPvp examples

* ScalingArmorPenetration example, typos, comments

* Add NPC.SuperArmor for damage calculations

* Update whip example to include a 'firecracker' single hit scaling bonus

* Change SuperArmor to override FinalDamage, make InstantKill do 1 damage

* Add a damage > 0 check to ExampleTeamDamageAccessory

* Add NPC.HideStrikeDamage

* Add a ModifyHitInfo callback backdoor for conversion effects

* Update advice in documentation

---------

Co-authored-by: JavidPack <javidpack@gmail.com>

direwolf420 committed:
ModTypeLookup.RegisterLegacyNames method (#3304)

* IModType.LegacyNames property

* rewrite to requested change

JavidPack committed:
GetItemDrops hook and redesigns to support Block Swap feature (#3210)

* GetItemDrops hook and redesigns to support Block Swap feature.

* Port remaining furniture to ItemDrop approach

* Remove ChestDrop/DresserDrop, use ItemDrop instead

* Port ExampleDresser, fix new features missing from ExampleChest, Fix #1981

* Make TileLoader.Drop consistent, add missing calls (Fix #1144 )

* Drop changes, automatic Tile/Style to item drop calculation, porter

* Port ExampleTrap, docs, AddContent example

* Fix ExampleDresser and ExampleTrap to use localization rework

* ItemDrop no longer needed in most situations. Address most feedback.

* Typo and comment fixes

* CanReplace hook.

* Tile replacement/removal set documentation

* More docs

* Fix for placing tile with no TileObjectData anchors

* TileLoader.Drop is now called once for multitiles and is consistent

* More comments, no single line if statements.

* Use placeStyle in chest placement failed fallback code

* Address some feedback

* Change GetItemDrops to IEnumerable<Item>

* Feedback.

* Docs and simplify logic

* remove allowFallbackToStyleZero option

* Docs, typo, note about pound tile

* tModPorter tests

* Update ModTileTest.Expected.cs

* Fix BasicChestFake tiles not calling usual modded drop methods

* Allow -1 values to indicate unconditional can't drop item state

* Support returning null in GetItemDrops

* Fix merge issues

Chicken-Bones committed:
Fix tModPorter UseType with arrays (for parameter refactors)

JavidPack committed:
Fix issue with RemoveSkippablePreview throwing exceptions

Chicken-Bones committed:
Replace tag get casts in WorldIO with defaulting get methods.

- Fixes crash when loading worlds with no npcTownVariationIndex on town NPCs

Chicken-Bones committed:
Lang file update for ExampleInfoAccessory

Solxanich committed:
Fix Unloaded Mod Accessory Slot #2667 (#3310)

* Fix UNloaded Mod Accessory Slot #2667

* Simplify AccessorySlotPlayer.LoadData

* Removed comment, netcode is correct

* Fix indexing issue when loaded slot count is > unloaded slot count

* Remove unnecessary UnloadedSlotCount

- Clearer invariants

* Add todo note to SaveData

---------

Co-authored-by: Chicken-Bones <mehvids@gmail.com>

JavidPack committed:
ContainerName changed to DefaultContainerName (#3311)

* ContainerName docs and porting. porting rule not working yet

* Rename ContainerName to DefaultContainerName

* Update Config.ModLoader.cs

blushiemagic committed:
Patron set for xAqult

JavidPack committed:
Typo

JavidPack committed:
Another typo

JavidPack committed:
Fix walls not dropping regression.

JavidPack committed:
Docs for methods showing text to player. Test include file approach

JavidPack committed:
HandlePacket clarification, Count/HasItem docs

Blockaroz committed:
Example Drill (#3271)

* Example Drill

* elaboration and accuracy

* mini refactor

* comment fix

* comment fix 2

* Example Drill finished

* Some fixes and cleanup for ExampleDrill, delete MoltenDrill from Old

---------

Co-authored-by: JavidPack <javidpack@gmail.com>

JavidPack committed:
RotatedRelativePoint, MountedCenter docs, fix Example Flail jitter

stormytuna committed:
Changes ModifyBuffTip hook into ModifyBuffText hook (#3307) (#3309)

* ModifyBuffTip -> ModifyBuffText

* Example mod changes

* Add tModPorter tests

* Remove comments

* Flip tip and rare back around

* Fix tests

* Add a buffName editing example

---------

Co-authored-by: JavidPack <javidpack@gmail.com>

JavidPack committed:
Localize ExampleGlobalBuff

Solxanich committed:
Update xAqult.cs

Solxanich committed:
Fix #3218

Solxanich committed:
Update xAqult.cs

JavidPack committed:
Port #2984 to 1.4.4. Fix #2984 (New hook for specifically item melee colliding)

Co-Authored-By: Creray <82579877+ItsCreray@users.noreply.github.com>

JavidPack committed:
RegisterKeybind doc update

JavidPack committed:
CanBeCollidedWithPlayerMeleeAttack->CanCollideWithPlayerMeleeAttack

Chicken-Bones committed:
Add ModifyHurtInfo and example of ModifyHitInfo

JavidPack committed:
Port ExampleExplosive, document related methods

JavidPack committed:
Comment out ExampleGlobalWall teaching example that is laggy.

Rijam committed:
Fix Town NPC Melee Swinging Source Rectangle (#3329)

* Updated DrawTownAttackSwing

* Updated summary docs for DrawTownAttackSwing

* Fixed typo in summary docs

* Keep texture2d, add rectangle ref

---------

Co-authored-by: JavidPack <javidpack@gmail.com>

absoluteAquarian committed:
Revert "Replace MonoMod Hooking of StackTrace with PathMap Usage in Projects (#3305)"

This reverts commit e4cb910feb455a6033290da30ac15e1b1aac6b1d.

absoluteAquarian committed:
Replace MonoMod Hooking of StackTrace with PathMap Usage in Projects (#3305)

* replace StackTrace hooking with PathMap usage

* remove `Logging.PrettifyStackTraceSources()` and its related members

absoluteAquarian committed:
improved the logic for prettifying StackTrace file paths to be more reliable (#3335)

JavidPack committed:
Update ExampleExplosive.cs

RegularPhoenix committed:
Ported Example Javelin (#3080)

* Added Example Javelin
Added Example Javelin Projectile

* Fixes, DoT, docs, glow, Old folder cleanup

---------

Co-authored-by: JavidPack <javidpack@gmail.com>

Fili-pk committed:
Fixed problem with detecting dotnet on linux (#3334)

* Fixed problem with detecting dotnet on linux

* Removed unnecessary semicolon

* Added old dotnet sdk check method as fallback

* Added check to make sure that dotnet path is valid

* Fixed typo

blushiemagic committed:
Patron sets for Frosty and Linus

Chicken-Bones committed:
Fix memory/performance leak on reload (#3035)

Chicken-Bones committed:
Unify Localized Conditions (#3341)

* Move Recipe.Condition to Condition. Improve Localizations

* Switch ItemVariants over to using Condition

* Add a way to convert a Condition to an IItemDropRuleCondition

* Add tModPorter

* Fix custom recipe condition example (and localization of)

* Improve example drop condition code. No need to check simulation

* Add don't starve seed condition, and tweak some naming

* Add some control/helpers for localizing SimpleItemDropRuleCondition

* Simplify ExampleNPCLoot examples

* Properly localize example drop conditions

* Condition -> Conditions in localizations

Chicken-Bones committed:
Fix typo

Chicken-Bones committed:
Rename file to match classname

Osjoud commented:
Pls help with #3342 for the love of god

BasicallyIAmFox committed:
Declarative Shops (#3219)

* yay

* changews

* Update Chest.cs.patch

* meow

* forgot

* Update TMLLootDatabase.cs

* Update TMLLootDatabase.cs

* uh

* Update ExamplePerson.cs

* A series of compilation fixes

* Final Compilation fixes

* Update TMLLootDatabase.cs

* Update TMLLootDatabase.cs

* Large Rework to Get it to Load in to a World for Testing

* Update TMLLootDatabase.cs

* Fixes for Invisible item issues

* Re-Add Pylons + Dummyproof EmptyInstance for SetDefaults

* queen and eol conditions

* painter left.

* painter

* Remove duplicate shop id conversion array + example global PostSetupShop

* Global NPC Test InsertAfter

* pylons

* trying to fix AddCandidates (№1)

* .

* Update ExampleGlobalNPC.cs

* should we change Entry to be class?

* Update ExampleBoneMerchant.cs

* updated parameters for SetupShop

* Update ExampleGlobalNPC.cs

* Update ChestLoot.Entry.cs

* removed old SetupShop hook and PylonShopNPC (not needed anymore hopefully)

* ModifyActiveShop hook

* Update NPCLoader.cs

* Update ExampleGlobalNPC.cs

* example mod fixes and expert/master conditions

* Update Chest.cs.patch

* changes

* removed OnFail and OnSuccess

* Review changes!

* Removed this[int] method from NPCShop

* Removed SetupLootboxShop and renamed TMLLootDatabase to NPCShopDatabase

* Funny changes! (too many to put them all in here)

* OrderLast, probably fixed SortBeforeAfter and added NoPylons dictionary

* readded PylonShopNPC, made sorting run after recipes

* guh

* Dread

* Mechanic is NOT Merchant!

* Accessory Slots are evil

* Moved DrawAccSlots call to be after shop rendering

* Added missing entries in the shops

* some documentation and tmodportertests (successful

* unbroke traveling merchant

* SetupShop is now ModifyActiveShop. BARTENDER

* Fix tModPorter tests, SetupShop -> ModifyActiveShop

* Better documentation and consistent parameter naming

* ExampleMod localization auto-export update

* Fix typo, Reverse -> Reserve

* NPCShop.items -> entries

* Improve NPCShop.Build code quality/performance

* Make NPCShop.Entry ordering calls more user friendly (like Recipe)

* Process NPCShop.Entry.OrdersLast before actual ordering

* Refactor shop sorting into its own method. Simplify

* Fix conditions in ExampleGlobalNPC

* Fix NPCShop.SortBeforeAfter so it actually sorts

* Move SortBeforeAfter into NPCShop (not entry)

* Simplify iteration in for ModifyShop

* Move bartender shop setup back into BartenderShopNPC

* Improve code formatting of Bartender shop

* Remove NPCShop. qualifiers everywhere in NPCShopDatabase

* Added Entry.OnShopOpened and made NPCShop.Build to clone items rather than directly passing them.

* Now Entry.OnShopOpen actually gets called whenever item is added to shop.

* Minor compile fixes and refactors

* Use a shop name for the travelling merchant's shop, for forwards compat

* Simplify SetupShop, always has an NPC reference. Use null for empty slots

* Add NPCShop.FillLastSlot

* Fix PostBuyItem being called twice

* More naming cleanup and property refactors

* Cleanup Insert/Add overloads, add InsertBefore/After(Entry)

* Fix brace formatting style

* Move NPCLoader shopToNpc into NPCShopDatabase

* doc fix

* Order pylons last

* Added condition for shimmered npcs. More in desc

- Added an example where item appears just if NPC is shimmered.
- Example where if NPC is shimmered, then all prices are reduces by 50%
- Moved shop examples from ExampleGlobalNPC to ExampleNPCShop
- :person_with_blond_hair:

* Cleanup example shop conditions

* Slight consistency/readability improvements for some moon phase conditions

* Use generic Add method in ExampleMod where possible

* Add Player.TalkNPC

* Fix wrong pylon conditions

* Improve pylon condition localisations

* Shop database cleanup, some condition fixes in there

* Keep vanilla setup shop method for comparison

* Fix NoPylons exclusion, meant to be travelling merchant

* Only make pylon entries once

* Add automatic shop testing code

* Fix several items, conditions and ordering issues in NPCShopDatabase

* Change ModPylon to use the new NPCShop.Entry system

* Improve localizations

* Further improve localizations

* Remaining localization key reference issues and naming tweaks

* Remove condition parameters from ModPylon.GetNPCShopEntry

* Fixed order of items in shops.

* More condition fixes as well as NoPylons now used

* Fixed compilation error

* Made pylon entries list to clear on unload (incase it doesn't already)

* Now using TryGetValue to avoid KeyNotFound exception

* Change NoPylons to be a Set

* Fix wrong condition for RoyalRomance

* Implement pylons for bartender with hack that matches vanilla

* Make shop test code ignore position rounding issues

* Add localizations for DownedQueenSlime condition

* Make shop overflow message less angry, and localize it

* Fix bartender pylon selling logic when > 1 pylon sold

* Revert the attempts to avoid condition quirks in shop tests

* Fix corruption and crimson localizations being reversed

* Use >= for golf score condition

* Add special steampunker outfit condition

* Disable testing code

* Call GetOrRegister when making conditions via the string ctor overload

* Merge NPCShop.Condition into Condition

* Slightly more consistent condition localizations

* Fix SetupTravelShop hook

* Add AbstractNPCShop and info entries for Travelling Merchant

* Rename Build -> FillShop. Add ICollection variant for future compat

* Rename AbstractNPCShop.IShopEntry to Entry

* Better ExampleTravelingMerchant showcasing new shop system

* Fix bug with ExampleTravelingMerchant

---------

Co-authored-by: nmoll <59670736+Solxanich@users.noreply.github.com>
Co-authored-by: Chicken-Bones <mehvids@gmail.com>

ExterminatorX99 committed:
update hooks (#3351)

Fili-pk committed:
Fixed problem with opening folders on linux (#3336)

Rijam committed:
More Conditions (#3348)

* More Conditions

* Added 3 more conditions

Destructor-Ben committed:
IL Editing QoL Stuff (#3252)

* Hook Dumping and LogILPatchFailure

* Added ILPatchFailureException

MutantWafflez committed:
Final Batch of 1.4.4 TODOs (#3350)

* Remove completed/ignored TODOs

* Preserve Modded Data in Journey Dupe UI

* BuyItemCreationContext w/ Example

* Revert tML publicization of WorldGen vars

* Small TODO comment update

* Remove redundant timePass variables

* Handful of ExampleOnBuyItem changes

* Item NewItem overload for dropItemCheck

* Update todo

* Remove Obsolete MessageIDs, tModPorter

* Remove redundant tab

* Deerclops dayRate change

* Unnecessary space removal

* RefactorInstanceMember to RenameStaticField

* Vanilla format for worldEventUpdates changes

* Various formatting changes, isAShopItem unset

* BuyItemCreationContext changes & EM changes

* NPCBoughtFrom -> VendorNPC

JavidPack committed:
Update ContentSamples.cs.patch

Fili-pk committed:
Fixed double damage space by trimming DisplayName of DamageType (#3349)

* Fixed double damage space by trimming DisplayName of DamageType

* Alternate approach

---------

Co-authored-by: JavidPack <javidpack@gmail.com>

Chicken-Bones committed:
Fix MonoModHooks.DumpIL implementation

Chicken-Bones committed:
Update ExampleMod localization files in other languages

Chicken-Bones committed:
Attempt to improve GetGlobal performance by removing unused arg

- inexact type lookups were low performance and rarely used. Can still be implemented manually.

Chicken-Bones committed:
Add tModPorter refactor for Condition.InGraveyard

Kogsey committed:
Fixes setting npcCatchable true forcing NPC.friendly to false (#3353)

* Fixes setting npcCatchable true forcing NPC.friendly to falce

* Fixed two possible coding practice sins

* Fix issue by fixing some vanilla npc values and adding a bool to track.

* Docs

* Comment

---------

Co-authored-by: JavidPack <javidpack@gmail.com>

Chicken-Bones committed:
Followup Improvements to StrikeNPC, Hit and Hurt hooks (#3355)

* Add max damage 'limit' to Hit/HurtModifiers

* Minor cleanup to CalculateStrikeFromLegacyValues

* Improve helper methods for simple StrikeNPC call use-cases

* Add documentation to Player StrikeNPC methods

* Fix inconsistent capitalization, `KnockBack` -> `Knockback`

* tModPorter for the couple of renames

* Reduce SuperArmor crit cap to 4

JavidPack committed:
ExampleJavelinBuff->ExampleJavelinDebuff

JavidPack committed:
Fix #3025

Chicken-Bones committed:
Provide HurtInfo argument to FreeDodge and ConsumableDodge (#3356) (#3359)

Chicken-Bones committed:
Fix default value of DefenseEffectiveness for NPCs (Fixes #3325)

Chicken-Bones committed:
Change some Main.max fields to static readonly in prep for future changes

Chicken-Bones committed:
Update MonoMod

Chicken-Bones committed:
Fix CheckBunny error in logs on non-windows platforms

JavidPack committed:
Change DrawTownAttackGun to support any texture

JavidPack committed:
Town AttackType examples: ExampleTravelingMerchant swings weapons, ExampleBoneMerchant shoots gun.

JavidPack committed:
Merge branch '1.4.4' into VerifyDotnetInstall
